controllers/rhmi/rhmi_controller.go

```diff
@@ -998,65 +979,63 @@ func (r *RHMIReconciler) preflightChecks(installation *rhmiv1alpha1.RHMI, instal
 		}
 	}
 
-	if rhmiv1alpha1.IsRHOAM(rhmiv1alpha1.InstallationType(installation.Spec.Type)) {
-		// Check if the quota parameter is found from the add-on
-		okParam, err := addon.ExistsParameterByInstallation(context.TODO(), r.Client, installation, addon.QuotaParamName)
+	// Check if the quota parameter is found from the add-on
+	okParam, err := addon.ExistsParameterByInstallation(context.TODO(), r.Client, installation, addon.QuotaParamName)
+	if err != nil {
+		preflightMessage := fmt.Sprintf("failed to retrieve addon parameter %s: %v", addon.QuotaParamName, err)
+		log.Warning(preflightMessage)
+		return result, err
```

Comment:
I'm confused on how is this to work with trialQuota. Without looking deeply into code, if quota param of `QuotaParamName` is not found, how can it ever attempt to check if `TrialQuotaParamName` is present if we are returning result,err here?

Reply:
This took bit of digging. The `okParam` is a boolean and is false if the parameter is not found in the addon secret. The `err` is nil if the parameter is not found in the secret. An error is only raised when there is a problem during the function call. Not finding the addon secret would raise an error for instances. This is ok and working as expected.

Reply:
Thanks for explanation, looks good